sokol_gfx.h

```diff
@@ -993,9 +1042,9 @@ typedef struct {
     uint32_t _start_canary;
     sg_pipeline pipeline;
     sg_buffer vertex_buffers[SG_MAX_SHADERSTAGE_BUFFERS];
-    uint32_t vertex_buffer_offsets[SG_MAX_SHADERSTAGE_BUFFERS];
+    int vertex_buffer_offsets[SG_MAX_SHADERSTAGE_BUFFERS];
```

Comment:
@floooh Out of curiosity, why do you change the type of the offsets from `uint32_t` to `int`?

Reply:
To make the integer types consistent across the API (e.g. the size attribute in sg_update_buffer and when creating resources are int).

Also I'm not yet sure what the return value of sg_append_buffer() should be when the buffer goes into "overflow" state (when the application tries to write more data then fits into the buffer). I *might* return a negative error value, but I'm immediately happy with that idea yet.